backend/ShootTheMoon/ShootTheMoon/Game/Game.cs

```diff
@@ -147,68 +148,119 @@ public IDisposable Subscribe(IObserver<GameEvent> observer)
             GameState previousState = State;
             State = newState;
 
-            Log.Debug("{0} => {1}", previousState, newState);
+            Log.Debug("{2}: {0} => {1}", previousState, newState, Name);
 
             if (State == GameState.DEALING) {
```

Comment:
Is it worth moving towards this sort of setup where we have a map of function pointers that is keyed based on the state?

https://stackoverflow.com/questions/44242575/static-map-to-function-pointers